src/main/java/duke/utils/Storage.java

```diff
@@ -0,0 +1,103 @@
+package duke.utils;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Scanner;
+
+import duke.exceptions.MemoryFailedException;
+import duke.tasks.Task;
+import duke.tasks.TaskType;
+
+/**
+ * Handles the loading of data from and storage of data to memory.
+ */
+public class Storage {
+
+    private Path memoryPath;
+    private File memory;
+
+    public Storage(String[] memoryPathArray) {
+        this.memoryPath = Paths.get(memoryPathArray[0],
+                Arrays.copyOfRange(memoryPathArray, 1, memoryPathArray.length));
+        this.memory = new File(String.valueOf(memoryPath));
+    }
+
+    /**
+     * Loads tasks from memory into the TaskList in the application.
+     *
+     * @param allTasks the TaskList containing all the tasks logged in by the user.
+     * @throws MemoryFailedException if there is an issue retrieving data from memory.
+     */
+    public void loadTasks(TaskList allTasks) throws MemoryFailedException {
+        try {
+            boolean success = this.memory.createNewFile();
```

Comment:
Perhaps an even more boolean name like `isSuccess`? I feel `success` is sufficient but maybe it could be even clearer although the variable is only used within this method.

Reply:
You are right, thank you for the suggestion!